docroot/modules/custom/mass_content/src/EventManager.php

```diff
@@ -16,11 +16,26 @@
  */
 class EventManager {
 
+  /**
+   * The relative date to use when building upcoming events.
+   *
+   * @var string
+   */
+  private $upcomingEndDateRelative;
+
   /**
    * Constructor.
    */
-  public function __construct(EntityTypeManagerInterface $entityTypeManager) {
+  public function __construct(EntityTypeManagerInterface $entityTypeManager, StateInterface $state) {
     $this->entityTypeManager = $entityTypeManager;
+    $this->upcomingEndDateRelative = 'today';
+
+    // $time is "now" only when running tests because we can't wait
+    // one day to test if the event is gone. Injecting the relative
+    // upcoming end date is complex, as well as changing the request time.
+    if ($state->get('UPCOMING_END_DATE_RELATIVE', FALSE)) {
+      $this->upcomingEndDateRelative = 'now';
+    }
```

Comment:
I'm not sure we want to use State directly in this class. Once we mock the time service, will this still be needed? It may be clearer to to have this string be a parameter in the constructor.

I can think further on this if I'm not making sense, let me know!